src/compiler/moduleSpecifiers.ts

```diff
@@ -170,8 +170,9 @@ namespace ts.moduleSpecifiers {
             return relativePath;
         }
 
-        const importRelativeToBaseUrl = removeExtensionAndIndexPostFix(relativeToBaseUrl, ending, compilerOptions);
-        const fromPaths = paths && tryGetModuleNameFromPaths(removeFileExtension(relativeToBaseUrl), importRelativeToBaseUrl, paths);
+        const bundledPkgReference = bundledPackageName ? combinePaths(bundledPackageName, relativeToBaseUrl) : relativeToBaseUrl;
+        const importRelativeToBaseUrl = removeExtensionAndIndexPostFix(bundledPkgReference, ending, compilerOptions);
+        const fromPaths = paths && tryGetModuleNameFromPaths(removeFileExtension(bundledPkgReference), importRelativeToBaseUrl, paths);

```

Comment:
I'm late to the party here, but I don't think it's right to pass the bundled package reference to `tryGetModuleNameFromPaths`. These values get tested for equality with the values in the `paths` value arrays (with `*` substitutions applied), which are absolute paths. `bundledPkgReference` will presumably not be an absolute path, so this will prevent `paths` from ever being used to generate a module specifier in tandem with `bundledPackageName`. Or, maybe that's the intent? I can't totally wrap my mind around how/why these options would ever be used together.